CRM/Relationshipblock/Utils/RelationshipBlock.php

```diff
@@ -23,11 +23,12 @@ public static function getExistingRelationships($contactID) {
       'contact_id_b' => $contactID,
       'return' => ['id', 'relationship_type_id', 'contact_id_a', 'contact_id_b',
         'contact_id_a.display_name', 'contact_id_b.display_name', 'contact_id_a.is_deceased',
-        'contact_id_b.is_deceased'],
+        'contact_id_b.is_deceased', 'end_date'],
       'options' => ['limit' => 0, 'sort' => 'relationship_type_id.label_a_b ASC', 'or' => [['contact_id_a', 'contact_id_b']]],
     ]);
     $ret = [];
     foreach ($existingRelationships['values'] as $rel) {
+      if ( !isset($rel['end_date']) || ($rel['end_date'] < $now )) {
```

Comment:
do you need to strtime these for more reliable comparison?

Reply:
Yes, I can use cron job "Disable expired relationships" for update "is_active" field for relationships, but this is I think better.

Reply:
@vitius1 I spotted a bug on this line. The variable `$now` has not been defined.

Reply:
For me it worked. But you can change for "strftime()". Maybe only on my website I have defined $now.

Reply:
According to [the docs](http://php.net/manual/en/function.strftime.php), you are using that function incorrectly by calling it with no params, as 1 param is required.

Reply:
@colemanw I'm sorry for this. I don't know well this function. But i try it, and it worked somehow. For correct syntax, it need to know the format of date for params. It works without params, but it's not correct.

Reply:
@colemanw I find the correct format of date. Here it is: strftime("%y-%m-%d").

Reply:
We don't usually use `strftime` in CiviCRM, I think it's more reliable to do 

    if (!isset($rel['end_date']) || (strtotime($rel['end_date']) < time() )) {

Reply:
@colemanw That was first time, when I worked with time in PHP and strftime was the first thing i found.